mesos-client/build.gradle

```diff
@@ -1,3 +1,11 @@
+sourceSets {
+    main {
+        scala {
+            srcDirs = ['src/main/scala', 'examples']
```

Comment:
for others I am including the slack discussion:

just think of this as a separate library. We might want to take this one day and put it into a separate repo for example because we will want to use it in marathon on somewhere else...
this is why I am trying to keep everything belonging to that library part of one module

I understood we are not doing the step of separating it and publishing as a jar dependency now because this way we could iterate on it while working on USI

Reply:
I think we need to define what the goal of this repo is, and what isn't.

The goal _is_, to have one implementation that can be used by all our orchestrators/frameworks.

The goal is _not_ to support any of the (related) components _outside_ of this project. I'm aware that implementations like persistence, metrics, leader election, mesos client, etc. are not necessarily part of the scheduler interface. They should be part of the surrounding tooling, and should for now be considered integral parts of this repository.

Please, don't plan with any of these implementations becoming standalone libraries. There is no use for us at this moment to go down that route. If we did, people would assume they're good to use as is, and we don't want to and cannot support someone using e.g. USI with their own persistence implementation, or using our Mesos client with their framework.